test cases/common/126 llvm ir and assembly/meson.build

```diff
@@ -46,9 +48,12 @@ foreach lang : ['c', 'cpp']
         input : square_preproc,
         output : lang + square_base + '.obj',
         command : [ml, '/Fo', '@OUTPUT@', '/c', '@INPUT@'])
-
   endif
-  e = executable('square_asm_' + lang, square_impl, 'main.' + lang,
-      c_args : uscore_args, cpp_args : uscore_args)
-  test('test ASM square' + lang, e)
+  if supported_cpus.contains(cpu)
+    e = executable('square_asm_' + lang, square_impl, 'main.' + lang,
+        c_args : uscore_args, cpp_args : uscore_args)
+    test('test ASM square' + lang, e)
+  else
+    message('Skipping square_asm test due to unsupported cpu: "' + cpu + '"')
```

Comment:
I guess we should skip this only when we're not using clang, because with clang we still run the LLVM IR test.